policy-controller/grpc/src/lib.rs

```diff
@@ -320,5 +343,195 @@ fn to_authz(
         networks,
         labels,
         authentication: Some(authn),
+        metadata: None, // TODO fill
     }
 }
+
+fn to_http_route(
+    name: impl ToString,
+    route: &HttpRoute,
+    srv: &InboundServer,
+    cluster_networks: &[IpNet],
+) -> proto::HttpRoute {
+    let metadata = Metadata {
+        kind: Some(metadata::Kind::Resource(
+            linkerd2_proxy_api::meta::Resource {
+                group: "gateway.networking.k8s.io".to_string(),
+                kind: "HTTPRoute".to_string(),
+                name: name.to_string(),
+            },
+        )),
+    };
+
+    let route = route.clone();
+
+    let hosts = route
+        .hostnames
+        .into_iter()
+        .map(|hostname| match hostname {
+            Hostname::Exact(host) => http_route::HostMatch {
+                r#match: Some(http_route::host_match::Match::Exact(host)),
+            },
+            Hostname::Suffix { reverse_labels } => http_route::HostMatch {
+                r#match: Some(http_route::host_match::Match::Suffix(
+                    http_route::host_match::Suffix {
+                        reverse_labels: reverse_labels.to_vec(),
+                    },
+                )),
+            },
+        })
+        .collect();
+
+    let authorizations = srv
+        .authorizations
+        .iter()
+        .map(|(n, c)| to_authz(n, c, cluster_networks))
+        .collect();
+    trace!(?authorizations);
```

Comment:
I think we probably ought to update the API so that we don't have to reserialize the same server authorizations on every route. Instead--for this PR--we'll return an empty array for routes. In a followup we'll set them from policy bindings.